collector/cluster_settings.go

```diff
@@ -151,5 +208,106 @@ func (c *ClusterSettingsCollector) Update(ctx context.Context, ch chan<- prometh
 		float64(shardAllocationMap[merged.Cluster.Routing.Allocation.Enabled]),
 	)
 
+	// Threshold enabled
+	thresholdMap := map[string]int{
+		"false": 0,
+		"true":  1,
+	}
+
+	ch <- prometheus.MustNewConstMetric(
+		clusterSettingsDesc["thresholdEnabled"],
+		prometheus.GaugeValue,
+		float64(thresholdMap[merged.Cluster.Routing.Allocation.Disk.ThresholdEnabled]),
+	)
+
+	// Watermark bytes or ratio metrics
+	if strings.HasSuffix(merged.Cluster.Routing.Allocation.Disk.Watermark.High, "b") {
+		ch <- prometheus.MustNewConstMetric(
+			clusterSettingsDesc["floodStageBytes"],
+			prometheus.GaugeValue,
+			getValueInBytes(merged.Cluster.Routing.Allocation.Disk.Watermark.FloodStage),
+		)
+
+		ch <- prometheus.MustNewConstMetric(
+			clusterSettingsDesc["highBytes"],
+			prometheus.GaugeValue,
+			getValueInBytes(merged.Cluster.Routing.Allocation.Disk.Watermark.High),
+		)
+
+		ch <- prometheus.MustNewConstMetric(
+			clusterSettingsDesc["lowBytes"],
+			prometheus.GaugeValue,
+			getValueInBytes(merged.Cluster.Routing.Allocation.Disk.Watermark.Low),
+		)
+
+		return nil
+	}
+
+	ch <- prometheus.MustNewConstMetric(
+		clusterSettingsDesc["floodStageRatio"],
+		prometheus.GaugeValue,
+		getValueAsRatio(merged.Cluster.Routing.Allocation.Disk.Watermark.FloodStage),
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		clusterSettingsDesc["highRatio"],
+		prometheus.GaugeValue,
+		getValueAsRatio(merged.Cluster.Routing.Allocation.Disk.Watermark.High),
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		clusterSettingsDesc["lowRatio"],
+		prometheus.GaugeValue,
+		getValueAsRatio(merged.Cluster.Routing.Allocation.Disk.Watermark.Low),
+	)
+
 	return nil
 }
+
+func getValueInBytes(value string) float64 {
```

Comment:
As a followup, this should return an error so that the metric is not reported and an error is logged when we fail to parse. There are also no tests for this function or the one below to make sure that no future changes break this functionality.